app/controllers/scorecards_controller.rb

```diff
@@ -4,14 +4,29 @@ class ScorecardsController < ApplicationController
   before_action :set_scorecard, only: [:show, :edit, :update, :destroy]
 
   def index
-    @pagy, @scorecards = pagy(
-      policy_scope(Scorecard.filter(filter_params)
-        .order("#{sort_column} #{sort_direction}")
-        .includes(
-          :facility, :local_ngo, :request_changes, :primary_school
+    respond_to do |format|
+      format.html {
+        @pagy, @scorecards = pagy(
+          policy_scope(Scorecard.filter(filter_params)
+            .order("#{sort_column} #{sort_direction}")
+            .includes(
+              :facility, :local_ngo, :request_changes, :primary_school
+            )
+          )
         )
-      )
-    )
+      }
+
+      format.xlsx {
+        @scorecards = policy_scope(Scorecard.filter(filter_params).order("#{sort_column} #{sort_direction}").includes(:participants, :raised_indicators, :voting_indicators))
+
+        if @scorecards.length > ENV.fetch("MAXIMUM_DOWNLOAD_RECORDS") { 1000 }.to_i
```

Comment:
@hengsokly can we move ENV.fetch("MAXIMUM_DOWNLOAD_RECORDS") { 1000 }.to_i to Setting.maximum_allowed_csv_records instead